bug: `RecordBone` in `RecordBone` is not handled

ViUR currently is not able to handle RecordBones in RecordBones.
```py
[2025-02-25 14:01:02,112] /.../viur/core/request.py:375 [ERROR] ViUR has caught an unhandled exception!
[2025-02-25 14:01:02,112] /.../viur/core/request.py:376 [ERROR] Read something from the datastore thats not a dict: <class 'viur.core.skeleton.SkeletonInstance'> <SkeletonInstance of subBoneRelation with {'subbone_name': 'blub'}> subrel
Traceback (most recent call last):
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/request.py", line 350, in _process
    self._route(path)
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/request.py", line 625, in _route
    res = caller(*self.args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/module.py", line 315, in __call__
    return self._func(self._instance, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/prototypes/list.py", line 280, in add
    skel.write()
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/skeleton.py", line 1460, in write
    bone.postSavedHandler(skel, bone_name, key)
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/bones/record.py", line 112, in postSavedHandler
    bone.postSavedHandler(using, bone_name, None)
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/bones/record.py", line 108, in postSavedHandler
    for idx, lang, value in self.iter_bone_value(skel, boneName):
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/bones/base.py", line 1385, in iter_bone_value
    value = skel[name]
            ~~~~^^^^^^
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/skeleton.py", line 264, in __getitem__
    boneInstance.unserialize(self, key)
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/bones/base.py", line 968, in unserialize
    res = self.singleValueUnserialize(loadVal)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/myproject/.venv/lib/python3.12/site-packages/viur/core/bones/record.py", line 67, in singleValueUnserialize
    assert isinstance(value, dict), f"Read something from the datastore thats not a dict: {type(value)} {value!r} {self.name}"
           ^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: Read something from the datastore thats not a dict: <class 'viur.core.skeleton.SkeletonInstance'> <SkeletonInstance of subBoneRelation with {'subbone_name': 'blub'}> subrel
```

skeletons/test.py:
```py
class subBoneRelation(RelSkel):
    subbone_name = StringBone(descr="name")


class simpleBoneRelation(RelSkel):
    relbone_name = StringBone(descr="name")
    subrel = RecordBone(
        descr="simple record bone",
        format="$(dest.name): $(dest.record_select)",
        using=subBoneRelation
    )


class RecTestSkel(Skeleton):
    name = StringBone(
        descr="Name",
        required=True,
    )

    simple_record = RecordBone(
        descr="simple record bone",
        format="$(dest.name): $(dest.record_select)",
        using=simpleBoneRelation
    )
```

modules/test.py:
```py
from viur.core.prototypes import List


class RecTest(List):
    pass
```

--- a/src/viur/core/bones/base.py
+++ b/src/viur/core/bones/base.py
@@ -559,6 +559,7 @@ class BaseBone(object):
         :return: A dictionary containing the collected raw client data.
         """
         fieldSubmitted = False
+
         if languages:
             res = {}
             for lang in languages:
@@ -584,7 +585,7 @@ class BaseBone(object):
                             if not key.startswith(prefix):
                                 continue
                             fieldSubmitted = True
-                            partKey = key.replace(prefix, "")
+                            partKey = key[len(prefix):]
                             firstKey, remainingKey = partKey.split(".", maxsplit=1)
                             try:
                                 firstKey = int(firstKey)
@@ -602,7 +603,7 @@ class BaseBone(object):
                             if not key.startswith(prefix):
                                 continue
                             fieldSubmitted = True
-                            partKey = key.replace(prefix, "")
+                            partKey = key[len(prefix):]
                             tmpDict[partKey] = value
                         res[lang] = tmpDict
             return res, fieldSubmitted
@@ -631,7 +632,7 @@ class BaseBone(object):
                         if not key.startswith(prefix):
                             continue
                         fieldSubmitted = True
-                        partKey = key.replace(prefix, "")
+                        partKey = key[len(prefix):]
                         try:
                             firstKey, remainingKey = partKey.split(".", maxsplit=1)
                             firstKey = int(firstKey)
@@ -649,7 +650,7 @@ class BaseBone(object):
                         if not key.startswith(prefix):
                             continue
                         fieldSubmitted = True
-                        subKey = key.replace(prefix, "")
+                        subKey = key[len(prefix):]
                         res[subKey] = value
                     return res, fieldSubmitted
 
@@ -1167,8 +1168,9 @@ class BaseBone(object):
         if query.queries and (orderby := params.get("orderby")) and utils.string.is_prefix(orderby, name):
             if self.languages:
                 lang = None
-                if orderby.startswith(f"{name}."):
-                    lng = orderby.replace(f"{name}.", "")
+                prefix = f"{name}."
+                if orderby.startswith(prefix):
+                    lng = orderby[len(prefix):]
                     if lng in self.languages:
                         lang = lng
 

--- a/src/viur/core/bones/record.py
+++ b/src/viur/core/bones/record.py
@@ -83,8 +83,7 @@ class RecordBone(BaseBone):
         if not value:
             return value
 
-        ret = value.serialize(parentIndexed=False)
-        return ret
+        return value.serialize(parentIndexed=False)
 
     def _get_single_destinct_hash(self, value):
         return tuple(bone._get_destinct_hash(value[name]) for name, bone in self.using.__boneMap__.items())

--- a/src/viur/core/bones/record.py
+++ b/src/viur/core/bones/record.py
@@ -64,7 +64,7 @@ class RecordBone(BaseBone):
         if isinstance(value, list) and value:
             value = value[0]
 
-        assert isinstance(value, dict), f"Read something from the datastore thats not a dict: {type(value)}"
+        assert isinstance(value, dict), f"Read {value=} ({type(value)})"
 
         usingSkel = self.using()
         usingSkel.unserialize(value)
@@ -83,7 +83,8 @@ class RecordBone(BaseBone):
         if not value:
             return value
 
-        return value.serialize(parentIndexed=False)
+        ret = value.serialize(parentIndexed=False)
+        return ret
 
     def _get_single_destinct_hash(self, value):
         return tuple(bone._get_destinct_hash(value[name]) for name, bone in self.using.__boneMap__.items())
@@ -97,27 +98,20 @@ class RecordBone(BaseBone):
 
     def singleValueFromClient(self, value, skel, bone_name, client_data):
         usingSkel = self.using()
+
         if not usingSkel.fromClient(value):
             usingSkel.errors.append(
                 ReadFromClientError(ReadFromClientErrorSeverity.Invalid, "Incomplete data")
             )
+
         return usingSkel, usingSkel.errors
 
     def postSavedHandler(self, skel, boneName, key) -> None:
         super().postSavedHandler(skel, boneName, key)
-        for idx, lang, value in self.iter_bone_value(skel, boneName):
-            using = self.using()
-            using.unserialize(value)
-            for bone_name, bone in using.items():
-                bone.postSavedHandler(using, bone_name, None)
 
-    def refresh(self, skel, boneName) -> None:
-        super().refresh(skel, boneName)
-        for idx, lang, value in self.iter_bone_value(skel, boneName):
-            using = self.using()
-            using.unserialize(value)
-            for bone_name, bone in using.items():
-                bone.refresh(using, bone_name)
+        for _, lang, value in self.iter_bone_value(skel, boneName):
+            for bone_name, bone in value.items():
+                bone.postSavedHandler(value, bone_name, None)
 
     def getSearchTags(self, skel: 'viur.core.skeleton.SkeletonInstance', name: str) -> set[str]:
         """
@@ -129,13 +123,14 @@ class RecordBone(BaseBone):
         """
         result = set()
 
-        using_skel_cache = self.using()
-        for idx, lang, value in self.iter_bone_value(skel, name):
+        for _, lang, value in self.iter_bone_value(skel, name):
             if value is None:
                 continue
-            for key, bone in using_skel_cache.items():
+
+            for key, bone in value.items():
                 if not bone.searchable:
                     continue
+
                 for tag in bone.getSearchTags(value, key):
                     result.add(tag)
 
@@ -178,11 +173,11 @@ class RecordBone(BaseBone):
         """
         result = set()
 
-        using_skel_cache = self.using()
-        for idx, lang, value in self.iter_bone_value(skel, name):
+        for _, lang, value in self.iter_bone_value(skel, name):
             if value is None:
                 continue
-            for key, bone in using_skel_cache.items():
+
+            for key, bone in value.items():
                 result |= bone.getReferencedBlobs(value, key)
 
         return result
@@ -198,16 +193,13 @@ class RecordBone(BaseBone):
     def structure(self) -> dict:
         return super().structure() | {
             "format": self.format,
-            "using": self.using().structure()}
+            "using": self.using().structure(),
+        }
 
     def refresh(self, skel, bone_name):
-        using_skel = self.using()
-
-        for idx, lang, value in self.iter_bone_value(skel, bone_name):
+        for _, lang, value in self.iter_bone_value(skel, bone_name):
             if value is None:
                 continue
 
-            using_skel.unserialize(value)
-
-            for key, bone in using_skel.items():
-                bone.refresh(using_skel, key)
+            for key, bone in value.items():
+                bone.refresh(value, key)

